vllm/executor/ray_gpu_executor.py

```diff
@@ -382,8 +432,17 @@ async def _run_task_with_lock(task, lock, *args, **kwargs):
         return results[-1]
 
     async def _start_worker_execution_loop(self):
+        assert not USE_SPMD_WORKER, (
+            "worker loop is disabled for VLLM_USE_SPMD_WORKER=1")
         coros = [
             worker.execute_method.remote("start_worker_execution_loop")
             for worker in self.non_driver_workers
         ]
         return await asyncio.gather(*coros)
+
+    def __del__(self):
+        if self.forward_dag is not None:
+            self.forward_dag.teardown()
+            import ray
+            for worker in self.workers:
+                ray.kill(worker)
```

Comment:
hmm this is a little hacky. Isn't this supposed to be guaranteed from .teardown API?

Reply:
`teardown()` method tears down the DAG, but not the actors -- I think the design was based on the idea that the actors can be reused (e.g., in another DAG).

Maybe later we can revisit the ADAG API, but with current version we need to eagerly kill them so Ray processes are cleaned up before next test runs.